src/lightning_fabric/strategies/strategy.py

```diff
@@ -62,6 +62,11 @@ def root_device(self) -> torch.device:
     def is_global_zero(self) -> bool:
         """Whether the current process is the rank zero process not only on the local node, but for all nodes."""
 
+    @property
+    @abstractmethod
+    def is_distributed(self) -> bool:
+        """Whether the strategy runs in a distributed environment."""
```

Comment:
There is a design where `isinstance(x, ParallelStrategy)` (and perhaps world_size>0) is equivalent with `is_distributed()=True`. In its current implementation, it is mainly needed to determine whether a special sampler is needed. There is also some outdated code in the TPU strategy. I believe long-term, this property would disappear again. 

I would consider keeping the PR as is, but remove it from the base class and do the `hasattr(self.strategy, "is_distributed")` call instead, so we don't have to deprecate it later on.

Reply:
@awaelchli the worldsize check would definitely be required, otherwise DP would falsely be marked as distributed as well (not sure we do support it with fabric today, but it would be a valid usecase). I think that I would like this API to exist, but later just return a check whether the collective is an actual collective or just the no-op one (which is what this ultimately comes to).

Reply:
I wouldn't go with `isinstance(x, ParallelStrategy)` because then it forces strategy authors to subclass it to correctly set the sampler.

A world size check would work, but maybe we want to keep `world_size==1` to return True for debugging purposes. For example, testing an interaction with the `DistributedSampler` with `strategy="ddp", devices=1`.

@justusschock Good catch with DataParallel, it needs to return `False` given the current use of `strategy.is_distributed`

Reply:
> I wouldn't go with isinstance(x, ParallelStrategy) because then it forces strategy authors to subclass it to correctly set the sampler.

Both ways, the developer would need to override and return the correct value. Maybe my point was not clear. I'm saying there is a design where there is no need for an attribute like this, where the distributed sampler creating is completely encapsulated by the strategy. I'm saying that, you can update the PR in a way that would allow us to later switch to this design without needing to deprecate anything.

Reply:
Okay. Actualyl we can use `hasattr(self.strategy, "distributed_sampler_kwargs")` which is already defined for the correct strategies